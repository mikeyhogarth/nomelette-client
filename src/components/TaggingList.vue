<template>
  <div>
    <h2>{{ title }}</h2>
    <ul>
      <li v-bind:key="tagging.slug" v-for="tagging in taggings">{{ tagging.name }}</li>
    </ul>
  </div>
</template>

<script>
export default {
  props: ["title", "taggings"]
};
</script>