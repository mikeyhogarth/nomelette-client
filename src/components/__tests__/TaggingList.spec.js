import { shallowMount, createLocalVue } from "@vue/test-utils";
import TaggingList from "@/components/TaggingList.vue";
import VueRouter from "vue-router";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("TaggingList", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(TaggingList, {
      localVue,
      propsData: {
        title: "Seasons",
        taggings: [
          { slug: "summer", name: "Summer" },
          { slug: "autumn", name: "Autumn" },
          { slug: "winter", name: "Winter" },
          { slug: "spring", name: "Spring" }
        ]
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
