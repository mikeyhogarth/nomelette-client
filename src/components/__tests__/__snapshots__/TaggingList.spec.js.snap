// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaggingList renders correctly 1`] = `
<div>
  <h2>
    Seasons
  </h2>
   
  <ul>
    <li>
      <router-link-stub
        event="click"
        tag="a"
        to="/tagged-with/summer"
      >
        
        Summer
      
      </router-link-stub>
    </li>
    <li>
      <router-link-stub
        event="click"
        tag="a"
        to="/tagged-with/autumn"
      >
        
        Autumn
      
      </router-link-stub>
    </li>
    <li>
      <router-link-stub
        event="click"
        tag="a"
        to="/tagged-with/winter"
      >
        
        Winter
      
      </router-link-stub>
    </li>
    <li>
      <router-link-stub
        event="click"
        tag="a"
        to="/tagged-with/spring"
      >
        
        Spring
      
      </router-link-stub>
    </li>
  </ul>
</div>
`;
